### DISCUSSION TITLE
model having type date has comparison issue using equals

### DISCUSSION THREAD

**Discussion by arpanbagui7** [prisma/prisma#14380] [Q&A]
My model contains a field like `calendar_date   DateTime     @db.Date`. Now in query I am trying to fetch records which match the same calendar_date but if I am writing like this 
`where: {
	calendar_date: dayjs(new Date()).format("YYYY-MM-DD"),
}` 
getting error  like "Provided String, expected DateTime" 
and if I tried like this 
`where: {
	calendar_date: new Date(),
}`
I am not getting any records in result as new Date() contains time also and calendar_date field has only date.
Can anyone tell me how to resolve this?

**Comment by ludralph** [comment]
Hi @arpanbagui7 👋,

At the moment, Prisma ORM doesn't have support for storing just Date. It has support for Datetime. However, there is a feature request for supporting just the Date format which can be seen [here](https://github.com/prisma/prisma/issues/4355). One workaround would be to pass the date returned from `dayjs` to a `Date()` object.
```js
where: { calendar_date: new Date(dayjs().format("YYYY-MM-DD")) }
```
It will return date of the form `2022-08-01T00:00:00.000Z`

**Reply by janpio** [reply]
Small correction @ludralph: It does support storing just Date, that is what you achieve with `DateTime @db.Date`. But Prisma CLient can not return just the Date part - it always returns a JS `Date` object which also includes the Time. Hence what is a `@db.Date` in the database with just `YYYY-MM-DD` always is and has to be a JS `Date` with `YYYY-MM-DDT00:00:00.000Z`. Then things should work.

**Reply by ludralph** [reply]
Thank you @janpio for the additional insights.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
